entry/src/main/java/com/thoughtbot/expandablerecyclerview/expand/ExpandAbilityHelper.java

```diff
@@ -56,194 +43,72 @@ public ExpandAbilityHelper(Context context, ComponentContainer rootView) {
-     */
-    private void prepareExpandableListAdapter() {
-        ExpandableListAdapter<String> expandableListAdapter = new ExpandableListAdapter<String>(context,
+    public void prepareExpandableListAdapter() {
+        ExpandableListAdapter<ParentChild> expandableListAdapter = new ExpandableListAdapter<ParentChild>(context,
                 mGroupNameItem, mGroupImageItem, ResourceTable.Layout_ability_listview_item) {
             @Override
-            protected void bind(ViewHolder holder, String text, Integer image, int position) {
-                if (!mTempChildNameItem.contains(text)) {
-                    // Set green background for parent/Group
-                    holder.makeInvisibleButton(ResourceTable.Id_checkbtn);
-                    holder.makeInvisibleImage(ResourceTable.Id_childstar);
-                    holder.setGroupItemBackground(ResourceTable.Id_groupContainer, ResourceTable.Color_white);
-                    holder.setText(ResourceTable.Id_tvGroupTitle, text, Color.GRAY,
-                            ResUtil.getIntDimen(context, ResourceTable.Float_group_text_size));
-                    holder.setGroupImage(ResourceTable.Id_ivGroupIcon, image,
-                            ShapeElement.RECTANGLE, Image.ScaleMode.STRETCH, ResourceTable.Color_white);
-                    if (!mTempGroupNameItem.contains(text)) {
-                        // Set divider & arrow down icon
-                        holder.setGroupImage(ResourceTable.Id_ArrowIcon, ResourceTable.Media_arrow_Down,
-                                ShapeElement.OVAL, Image.ScaleMode.CENTER, ResourceTable.Color_white);
-                    } else {
-                        // Remove divider & arrow up icon
-                        holder.setGroupImage(ResourceTable.Id_ArrowIcon, ResourceTable.Media_arrow_Up,
-                                ShapeElement.OVAL, Image.ScaleMode.CENTER, ResourceTable.Color_white);
-                    }
-                } else {
-                    // Add child items to list
-                    holder.makeInvisibleButton(ResourceTable.Id_checkbtn);
-                    holder.makeInvisibleImage(ResourceTable.Id_checkbtn);
-                    holder.setText(ResourceTable.Id_tvGroupTitle, text, Color.GRAY,
-                            ResUtil.getIntDimen(context, ResourceTable.Float_child_text_size));
-                }
+            protected void bind(ViewHolder holder, ParentChild text, Integer image, int position) {
+                handleListItem(holder, text, image);
             }
         };
-        mGroupContainer.setItemProvider(expandableListAdapter);
-
-        expandableListAdapter.setOnItemClickListener((component, position) -> {
-            String clickedItem = mGroupNameItem.get(position);
-            checkChild(clickedItem, expandableListAdapter);
-        });
+        //setting the adapter and the Clicklistener
+        setterfunction(expandableListAdapter);
```

Comment:
This code is also duplicated in multiple helper files. Can be moved to parent class